admin-tools/build_and_check_manifest.py

```diff
@@ -10,15 +10,8 @@ def generate_avaliable_builtins_names():
     for module in modules:
         vars = dir(module)
         for name in vars:
-            var = getattr(module, name)
-            if (
-                hasattr(var, "__module__")
-                and var.__module__.startswith("mathics.builtin.")
-                and var.__module__ != "mathics.builtin.base"
-                and is_builtin(var)
-                and not name.startswith("_")
-                and var.__module__ == module.__name__
-            ):  # nopep8
+            var = contributing_builtin_var(module, name)
```

Comment:
Using WMA docs, they do not use the name "variable" (or var), but _symbol_.  And this matches our terminology used elsewhere too: we have a file called `SYMBOLS_MANIFEST` (not`VAR_MANIFEST`). 

The other aspect of this function name I find jarring is the use of the verb form "contributing" rather than active verb "contributes". (We saw this tense thing previously in summary text). 

If you run `git grep -n 'def .*ing.*('` you'll see very few functions that have use a verb in this tense. Here are some matches of that search: 
  * `decreasing`  would have been better named `is_descreasing` since this is a boolean function
  * `find_matching_indices...` `reduce_series_trailing_zeros`, etc. These are adjectives, not verb tenses.
  * `to_python_encoding`, `_apply_boxscaling`, `_boxscaling`, `padding`, etc.  These are nouns. 

But even changing `contributing_builitin_var` to `contributes_symbol` there still feels something jarring. I think it is that the use of the word "contribute" instead of a simple word like "add" and the vagueness of the name: contibutes _what?_

The intent I think is to have a collection of symbols or a list of Built-in Functions and Mathics Symbols that are not user visible and neither appear in documentation nor `SYMBOLS_MANIFEST.txt`

Given this, an alternate name change is `adds_visible_symbol` and `NO_CONTRIBUTION_CLASSES` might be `NOT_VISIBLE_SYMBOL`. 

Why do I spend such time on stuff that seems rather small? 

For me, a lot of the ugliness of this code base stems from the imprecision of intent. This vagueness allows misunderstanding, and then after others then add to it (or sometimes even if not),the resulting code is both sloppy _and_ inefficient. 

If earlier on a clearer distinction were made between expression evaluation and boxing, I wonder if the messy code in `Expression` would have been so complicated _and_ so inefficient. But some of the inefficiency in Expression was also not a clear distinction between a String, and a Lisp symbol.

Reply:
> Using WMA docs, they do not use the name "variable" (or var), but _symbol_. And this matches our terminology used elsewhere too: we have a file called `SYMBOLS_MANIFEST` (not`VAR_MANIFEST`).
>

I have use this name just because the lhs is "var" and the function in which it is used is named "contribute". This just replaces a more or less long set of checks to see if the object with name "name" in the module "module" should be added as a definition.
 
> The other aspect of this function name I find jarring is the use of the verb form "contributing" rather than active verb "contributes". (We saw this tense thing previously in summary text).
> 
> If you run `git grep -n 'def .*ing.*('` you'll see very few functions that have use a verb in this tense. Here are some matches of that search:
> 
>     * `decreasing`  would have been better named `is_descreasing` since this is a boolean function
> 
>     * `find_matching_indices...` `reduce_series_trailing_zeros`, etc. These are adjectives, not verb tenses.
> 
>     * `to_python_encoding`, `_apply_boxscaling`, `_boxscaling`, `padding`, etc.  These are nouns.
> 
> 
> But even changing `contributing_builitin_var` to `contributes_symbol` there still feels something jarring. I think it is that the use of the word "contribute" instead of a simple word like "add" and the vagueness of the name: contibutes _what?_
> 
> The intent I think is to have a collection of symbols or a list of Built-in Functions and Mathics Symbols that are not user visible and neither appear in documentation nor `SYMBOLS_MANIFEST.txt`
> 
> Given this, an alternate name change is `adds_visible_symbol` and `NO_CONTRIBUTION_CLASSES` might be `NOT_VISIBLE_SYMBOL`.
>

OK.
 
> Why do I spend such time on stuff that seems rather small?
> 
> For me, a lot of the ugliness of this code base stems from the imprecision of intent. This vagueness allows misunderstanding, and then after others then add to it (or sometimes even if not),the resulting code is both sloppy _and_ inefficient.
> 
> If earlier on a clearer distinction were made between expression evaluation and boxing, I wonder if the messy code in `Expression` would have been so complicated _and_ so inefficient. But some of the inefficiency in Expression was also not a clear distinction between a String, and a Lisp symbol.

Reply:
@rocky, thinking again about this, `contributing_builtin_var`  does not add anything, just checks if certain name should be added as a definition, and returns the associated `Builtin` class. Said that, pick the name that sounds more suitable to you.